packages/core/test/types.test.ts

```diff
@@ -480,3 +482,17 @@ describe('interpreter', () => {
     });
   });
 });
+
+describe('spawn', () => {
+  it('spawned actor ref should be compatible with the result of ActorRefFrom', () => {
+    const createChild = () => createMachine({});
+
+    function createParent(_deps: {
+      spawnChild: () => ActorRefFrom<typeof createChild>;
```

Comment:
The issue was that in here we are unpacking the type from a factory but `spawn` below actually unpacks from a machine (since the result of a factory is passed to it). Since resolving from factories and types was not unified... it caused a type mismatch.